@here/harp-mapview/test/PoiRendererTest.ts

```diff
@@ -97,4 +100,93 @@ describe("PoiRenderer", function () {
+        const webGLRenderer = {
+            capabilities: {
+                isWebGL2: false
+            }
+        } as THREE.WebGLRenderer;
+        const mapViewStub = sinon.createStubInstance(MapView);
+        const poiManager = new PoiManager((mapViewStub as any) as MapView);
+        const testImageName = "testImage";
+        const mapEnv = new Env();
+
+        const createPointLabel = (name: string) => {
+            return {
+                poiInfo: {
+                    imageTextureName: name,
+                    isValid: true,
+                    buffer: undefined,
+                    technique: {}
+                },
+                visible: true
+            } as TextElement;
+        };
+
+        let x = 1;
+        let y = 1;
+
+        const addRandomImageToCache = (
+            testCache: MapViewImageCache,
+            name: string,
+            load: boolean
+        ) => {
+            // Note, the images must be unique, otherwise the test will fail, because the internal
+            // caching mechanism of the ImageCache will have already loaded the image.
+            return testCache.addImage(name, `https://picsum.photos/${x++}/${y++}`, load);
+        };
+
+        it("poi is valid when in cache and loading started", async () => {
+            const testCache = new MapViewImageCache();
+            const testImageItem = addRandomImageToCache(testCache, testImageName, true);
+            const caches = [testCache];
+            const poiRenderer = new PoiRenderer(webGLRenderer, poiManager, caches);
+            const pointLabel = createPointLabel(testImageName);
+            const waitingForLoad = poiRenderer.prepareRender(pointLabel, mapEnv);
+            // This is false because the image is still being loaded in the background, notice the
+            // true flag passed to the call to testCache.addImage.
+            expect(waitingForLoad).false;
+
+            // Promise.resolve used to convert the type `ImageItem | Promise<ImageItem>` to
+            // `Promise<ImageItem>`.
+            await Promise.resolve(testImageItem);
+
+            const imageLoaded = poiRenderer.prepareRender(pointLabel, mapEnv);
+            expect(imageLoaded).true;
+        });
+
+        it("poi is valid when in cache and loading not started", async () => {
+            const testCache = new MapViewImageCache();
+            // This is the difference to the test above, we don't start loading, we expect the
+            // prepareRender function to do this.
+            addRandomImageToCache(testCache, testImageName, false);
+            const caches = [testCache];
+            const poiRenderer = new PoiRenderer(webGLRenderer, poiManager, caches);
+            const pointLabel = createPointLabel(testImageName);
+            const waitingForLoad = poiRenderer.prepareRender(pointLabel, mapEnv);
+            const result = await Promise.resolve(waitingForLoad);
+            expect(result).true;
```

Comment:
this is the test which required the change to return a Promise<boolean>

Reply:
If you added the promise just for testing, there's another option that would leave the production code intact. You can wait for the async tasks to be executed doing:

`const wait = (ms: number = 0) => new Promise(res => setTimeout(res, ms));

await wait();
`
It's used in TileGeometryLoaderTest, TileGeometryCreatorTest and DecodedTilesHelperTest.

Reply:
As discussed, I tried this, but it doesn't work, so will resolve.

Reply:
@nzjony You missed this comment.